### PR TITLE
Actualization common

### PR DESCRIPTION
**Pull request by kosteman**
Список изменений:
- Полностью завершена актуализация. 
- Зарефакторены парсеры капибилити через интерфейсы

### CODE REVIEW

File: java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/SelenoidCapabilitiesParser.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            case IE:
+                handleIe();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void handleOpera() {
+        capabilities.setCapability("operaOptions", new HashMap<String, String>() {
```
Comment (varivoda) [inline java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/SelenoidCapabilitiesParser.java]: MINOR Use another way to initialize this instance. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3599)

File: java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/SelenoidCapabilitiesParser.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            {
```
Comment (varivoda) [inline java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/SelenoidCapabilitiesParser.java]: MAJOR Move the contents of this initializer to a standard constructor or to field initializers. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1171)

File: java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/WebDriverCapabilitiesParser.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    public DesiredCapabilities parse() {
+        Set<String> properties = Props.getProps().stringPropertyNames();
+        List<String> capabilitiesWithPrefix = getCapabilitiesWithPrefix(properties, CAPABILITY_WITH_PREFIX_REGEX);
+
+        for (String capabilityWithPrefix : capabilitiesWithPrefix) {
+
+            String capabilityName = cutPrefix(capabilityWithPrefix);
+            String capabilityValue = Props.get(capabilityWithPrefix);
+
+            switch (TagWebDriver.getBrowserName()) {
```
Comment (varivoda) [inline java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/WebDriverCapabilitiesParser.java]: MINOR Replace this "switch" statement by "if" statements to increase readability. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)

File: java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/WebDriverCapabilitiesParser.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                case CHROME:
```
Comment (varivoda) [inline java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/capabilities/WebDriverCapabilitiesParser.java]: BLOCKER End this switch case with an unconditional break, return or throw statement. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS128)

**Comment by varivoda** [comment]
SonarQube analysis reported 5 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PageReflectUtil.java#L365](https://github.com/sbtqa/page-factory-2/blob/cc842d323af12ff95b40b408c26dbf0047ec1b29/java/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/PageReflectUtil.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/WebElementsPage.java [hunk 5]
```diff
@@ -116,7 +117,7 @@ public void clickElementByTitle(String elementTitle) throws PageException {
         try {
             webElement = PageFactoryUtils.getElementByTitle(PageContext.getCurrentPage(), elementTitle);
             DriverExtension.waitForElementGetEnabled(webElement, PageFactory.getTimeOut());
-        } catch (NoSuchElementException | WaitException e) {
+        } catch (NoSuchElementException | WaitException | ElementNotFoundException e) {
```
Comment (0I0u3uk) [inline java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/WebElementsPage.java]: Это должно быть в плагине пофикшено, WebElementPage из core был удален, не знаю почему он в master есть, я в своем pull request его еще раз удаляю

**Comment by varivoda** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PageReflectUtil.java#L365](https://github.com/sbtqa/page-factory-2/blob/71e4f4c3f8be4f922411c823a5dbf8cae8128b6e/java/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/PageReflectUtil.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by varivoda** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PageReflectUtil.java#L365](https://github.com/sbtqa/page-factory-2/blob/5245844a4e1c3679b4df7913a795730dadcd84ba/java/plugins/html-elements-plugin/src/main/java/ru/sbtqa/tag/pagefactory/PageReflectUtil.java#L365): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**